### PR TITLE
Add reminder summary widget

### PR DESCRIPTION
**Pull request by elcct**
## Description of change

This adds a reminder summary widget for personalised dashboard.

It requires user to have `reminder-summary` feature flag.

## Test instructions

Go to personalised dashboard, you should be able to see the widget on the left hand side, provided you have `reminder-summary` user feature flag set.

### Before

reminders-before

### After

reminders-after

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/client/components/ReminderSummary/Summary.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+              .map(([type, number]) => (
+                <StyledListItem key={type} data-test={`summary-item-${type}`}>
+                  <StyledReminderLink href={typeToURL[type]}>
+                    {typeToReminderName[type]}
+                  </StyledReminderLink>
+                  &nbsp;({number})
+                </StyledListItem>
+              ))}
+          </StyledList>
+          <br />
```
Comment (Harriethw) [inline src/client/components/ReminderSummary/Summary.jsx:64]: minor: are these `<br />`s to add white space? would it be possible to create a space here (ideally using the `@govuk-react/constants` SPACING constants? `<br /> have a semantic meaning in HTML to create line breaks in text.

File: test/functional/cypress/specs/dashboard-new/reminder-summary-spec.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      cy.get('@notificationBadge').should('have.text', '8')
+    })
+
+    it('should contain elements in the correct order', () => {
+      cy.get('@reminderSummary').contains(
+        'Approaching estimated land dates (1)Projects with no recent interaction (2)Outstanding propositions (5)Reminders and email notifications settings'
+      )
+      /*
+      cy.get('@reminderSummary').should(
+        'have.text',
```
Comment (Harriethw) [inline test/functional/cypress/specs/dashboard-new/reminder-summary-spec.js:52]: was this commented out text accidentally left in?

File: src/client/components/ReminderSummary/tasks.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { apiProxyAxios } from '../Task/utils'
+
+export const fetchReminderSummary = () =>
+  apiProxyAxios.get('/v4/reminder/summary').then(({ data }) => data)
```
Comment (Harriethw) [inline src/client/components/ReminderSummary/tasks.js:4]: will there be any error handling?

File: src/client/components/ReminderSummary/Summary.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        </>
+      )}
+      {!count && (
```
Comment (Harriethw) [inline src/client/components/ReminderSummary/Summary.jsx:67]: minor/opinion: would this make sense as an if/else or ternary? if count is bigger than 0 then show the summary, if not show no reminders.

**Review by Harriethw — APPROVED** [review]
looks mostly good to me :)

**Comment by paulgain** [comment]
This may be a question for UX but I think we should always have those links visible. If a user has no outstanding propositions we shouldn't remove the link, we should display the link as 'Outstanding propositions (0)'. The same applies to both 'Approaching estimated land dates' and 'Projects with no interaction'

File: src/client/components/ReminderSummary/tasks.js (hunk 3 above)
Comment (paulgain) [reply src/client/components/ReminderSummary/tasks.js:4]: The error handling is baked into `apiProxyAxios`

File: src/client/components/ReminderSummary/Summary.jsx (hunk 1 above)
Comment (elcct) [reply src/client/components/ReminderSummary/Summary.jsx]: Of course I should have done SPACING. Fixed now. Thank you!

File: test/functional/cypress/specs/dashboard-new/reminder-summary-spec.js (hunk 2 above)
Comment (elcct) [reply test/functional/cypress/specs/dashboard-new/reminder-summary-spec.js]: Yes! Thank you!

File: src/client/components/ReminderSummary/Summary.jsx (hunk 4 above)
Comment (elcct) [reply src/client/components/ReminderSummary/Summary.jsx]: I updated it with ternary. It's better! Thank you.

**Comment by elcct** [comment]
@paulgain I can ask, but the designs show the link should disappear if there is no reminders for the particular type.

**Pull request merged**